### DISCUSSION TITLE
2.17.1-1

### DISCUSSION THREAD

**Discussion by bb-qq** [bb-qq/r8152#317] [Announcements]
⚠️If you want to report a problem or get support, **please fill out the template and create a ticket in [Issues](https://github.com/bb-qq/r8152/issues)** instead of posting in the forum.

* The base code has been updated to version 2.17.1 which includes the following changes:
  * fix the autosuspend doesn't work
  * move setting r8153b_rx_agg_chg_indicate()
  * fix the poor throughput for 2.5G devices
  * fix flow control issue of RTL8156A
  * Add __GFP_NOWARN to big allocations
  * reduce the control transfer of rtl8152_get_version()
  * r8152: avoid to change cfg for all devices
 
Note:

* The installation will fail the first time. After that, run the following command from the SSH terminal:
   `sudo install -m 4755 -o root -D /var/packages/r8152/target/r8152/spk_su /opt/sbin/spk_su`
* Selecting a proper version is needed depending on your DSM version.
  * DSM 7.2 and above
    * use packages with the suffix `_7.2`.
  * DSM 7.0 and DSM 7.1
    * use packages with the suffix `_7.1`.
  * DSM 6.2.x
    * use packages with the suffix `_6.2`.

<hr /><em>This discussion was created from the release <a href='https://github.com/bb-qq/r8152/releases/tag/2.17.1'>2.17.1-1</a>.</em>

**Comment by PigSnack** [comment]
Still at sub 700 Mb speeds on 2.5 Gb nics. Updated to _7.2 on DSM 7.2. sudo install from SSH and reboot.

**Reply by PigSnack** [reply]
Issue is with MB integrated Realtek 2.5 Gbe nic driver.  I obtained a Realtek USB 2.5 nic for PC endpoint and issue resolved.  The problem started with a Win 11 update. I previously had  no issue with integrated Realtek 2.5 Gbe nic and Realtek USB at NAS running 2.16.

**Reply by wilson-9287** [reply]
@bb-qq Hi brother, may i know the driver support DSM 7.2.1? Thanks

**Reply by PigSnack** [reply]
2.17.1-1 is working fine for me with DSM 7.2.1-69057

**Reply by wilson-9287** [reply]
Thanks! let me try again

**Comment by bwisely** [comment]
In this version, my DS1815+ cannot be loaded normally. You can see that the initial is successful, but the network card often does not see it, or cannot get the IP address (DHCP mode), and the manual setting (static mode) also does not work.
by the way, I tried to ping the USB NIC, I can see the MAC address on my PC side, but no response.
The strange thing is that the link speed is only 100M
 (This USB NIC, cable moved to my PC. the link speed of 1000M. )

USB NIC=RTL8153, tplink
DSM=DSM 7.1.1-42962 Update 6
CPU=INTEL Atom C2538


2.16.3-4 on [ 7.1.1 update 5 ]  works well with a link speed of 1000M.

**Reply by bb-qq** [reply]
Did the previous version of the driver work fine?

**Reply by bwisely** [reply]
previous version cannot work on [DSM 7.1.1-42962 Update 6]
but 2.16.3-4 on [ 7.1.1 update 5 ] is ok

**Reply by bb-qq** [reply]
Hmm, could you try [full reinstallation](https://github.com/bb-qq/r8152/wiki/Troubleshooting#uninstall-completely--full-reinstallation)?

**Reply by bwisely** [reply]
yes, I followed your suggestion and did the steps. and re-install again. the issue is same the as before.
the interface is ready, but Synology Assistant cannot find the interface, and my DS1815 cannot through this USB interface to access the internet.

**Reply by bb-qq** [reply]
There are various changes on Realtek's side in 2.17, including firmware, which may have introduced the bug.
I'm sorry, but I don't know any more details than that.

**Reply by bwisely** [reply]
thanks and if the issue can be solved. it's great. if not, i need to change back to the default.
hope to hear good news.

**Comment by activebeergeek** [comment]
Hello!

I'm trying to get my 2.5gbe USB adaptor to work with my DS1817+.

I tried to install r8152-avoton-2.17.1-1_7.2.spk but the install fails on the Synology web interface when I manually install it. I can see the correct app in the list of installed packages, but there is a red error.

I'm on DSM 7.2-64570 Update 3

**Reply by bb-qq** [reply]
Did you follow the instructions on the README?
If the problem persists, please fill out the template and create a ticket in [Issues](https://github.com/bb-qq/r8152/issues).

**Comment by remichapeaublanc** [comment]
First of all, thank you for this amazing job.

I was more than happy with this drivers, using it on a DS1621+ and a AS-U2.5G2.
But I bought a new AS-U2.5G2 to put 2x 2,5Gbs on my Synology NAS (DSM 7.2.1-69057 Update 1) to enjoy SMB3 multichanel (with a 2x 2,5Gbs on my Mac too).
But when I plug the second AS-U2.5G2, it show me the new LAN connection, but forget the older one.
Any idea how to resolve this issue?

**Reply by bb-qq** [reply]
Please check this page.
https://github.com/bb-qq/r8152/wiki/Troubleshooting#multiple-identical-devices-do-not-work

**Comment by UNIPART82** [comment]
![SA6400-Synology-NAS (6)](https://github.com/bb-qq/r8152/assets/162960224/782c896e-d9ad-4626-8491-e420556b69c1)
![SA6400-Synology-NAS (5)](https://github.com/bb-qq/r8152/assets/162960224/d5f98700-2f5c-4b12-9661-4a5604e761fa)
![pve-Proxmox-Virtual-Environment (18)](https://github.com/bb-qq/r8152/assets/162960224/fa78402c-83f7-4dea-adb3-1c990895b390)
![pve-Proxmox-Virtual-Environment (17)](https://github.com/bb-qq/r8152/assets/162960224/6ab2f99b-af41-4853-8e0b-422aabd6fda1)


CPU : Intel I3-12100
Proxmox VE : 100 (NAS)
Model : SA6400
Ver : DSM 7.2.1
What driver should I download for using 2.5G USB LAN?

**Reply by bb-qq** [reply]
I recently added the driver for SA6400. #382

**Comment by bgood66** [comment]
Hello thanks for this drivers its worked well on my ds1520+ with this https://www.amazon.fr/dp/B07Q626XK2?psc=1&ref=ppx_yo2ov_dt_b_product_details

but the last DSM 7.2.1-69057 Update 5 Killed it the driver don't start.
Hope you can fix this  
![image](https://github.com/bb-qq/r8152/assets/17361285/fd5aa185-bcdf-4439-b10c-3d657f2466ef)

**Reply by bb-qq** [reply]
Could you fill out the template and create a ticket in [Issues](https://github.com/bb-qq/r8152/issues)?
Also, this page may help.
https://github.com/bb-qq/r8152/wiki/Troubleshooting#how-to-investigate-if-the-driver-does-not-work-properly

**Comment by johsaj** [comment]
USB NIC = AX88179A
Cable Matters 2.5G USB 3.0 to Gigabit LAN https://www.amazon.ca/gp/product/B013G4C8RE
DSM = DSM 7.1.1-42962 Update 6
CPU = INTEL Atom C2538
RAM = 8192MB
Synology model = Synology 1515+
r8152-avoton-2.17.1-1_7.1.spk installed and with the readme file “sudo install” command assistance, the LAN 5 (as seen in the DSM GUI) is operating at 1000Mbs, Full Duplex, with standard MTU of 1500.

This worked from Jan 24, 2024 to Feb 8, 2024 when all USB ports crashed within an hour of each other. The USB 8TB drive was doing a backup that had already been running for 4 days but the drive disconnected. Then the 5TB drive started to do its backup but it disconnected, and at the same time the UPS (USB link to UPS) and LAN 5 disconnected. Unplug and replug in of USB cables did not get NAS to see the drives, LAN 5, or the UPS connection.

A reboot of NAS did not help to restore USB. Rebooted NAS and it could see the UPS and the USB drives but I had to stop and start the RTL8152/RTL8153 bb-qq package to get the LAN 5 to be seen.

USB network has been working fine since Feb 9, 2024 00:26am using r8152-avoton-2.17.1-1_7.1.spk. Haven’t been doing speed tests but regularly see speeds (as reported by DSM GUI) of 950Kb/s - using that network for the Synology Surveillance out of band camera network. A few weeks after that, the 8TB drive failed completely and was replaced with a new 14TB drive.

On Dec 21, 2024, I upgraded to r8152-avoton-2.18.1-2_7.1.spk, and then upgraded DSM to DSM 7.1.1-42962 Update 7. No issues so far.
